Sources/Atomic.swift

```diff
@@ -122,33 +218,16 @@ internal final class RecursiveAtomic<Value>: AtomicProtocol {
 	///
 	/// - returns: The result of the action.
 	@discardableResult
+	@inline(__always)
 	func withValue<Result>(_ action: (Value) throws -> Result) rethrows -> Result {
 		lock.lock()
-		defer { lock.unlock() }
-
-		return try action(_value)
-	}
-}
-
-public protocol AtomicProtocol: class {
-	associatedtype Value
-
-	@discardableResult
-	func withValue<Result>(_ action: (Value) throws -> Result) rethrows -> Result
-
-	@discardableResult
-	func modify<Result>(_ action: (inout Value) throws -> Result) rethrows -> Result
-}
-
-extension AtomicProtocol {	
-	/// Atomically get or set the value of the variable.
-	public var value: Value {
```

Comment:
The performance is much worse due to the need to query the protocol witness table, even with the `@inline(__always)` annotation. I have rerun my benchmarks and it is still the case.

I am actually thinking of removing `AtomicProtocol`, hmm...

> Embed PosixThreadMutex into Atomic.
> 0.179 sec (4%)
>
> Implement value and swap in Atomic.
> 0.149 sec (4%)